specification/resources/resource-manager/Microsoft.Resources/stable/2024-03-01/deploymentStacks.json

```diff
@@ -0,0 +1,1652 @@
+      ],
+      "description": "Entity representing the reference to the deployment parameters."
+    },
+    "DeploymentStacksDebugSetting": {
+      "type": "object",
+      "properties": {
+        "detailLevel": {
+          "type": "string",
+          "description": "Specifies the type of information to log for debugging. The permitted values are none, requestContent, responseContent, or both requestContent and responseContent separated by a comma. The default is none. When setting this value, carefully consider the type of information that is being passed in during deployment. By logging information about the request or response, sensitive data that is retrieved through the deployment operations could potentially be exposed."
+        }
+      },
+      "description": "The debug setting."
+    },
+    "DeploymentStacksDeleteDetachEnum": {
+      "type": "string",
+      "enum": [
+        "delete",
+        "detach"
+      ],
+      "description": "Specifies an action for a newly unmanaged resource. Delete will attempt to delete the resource from Azure. Detach will leave the resource in it's current state.",
+      "x-ms-enum": {
+        "name": "DeploymentStacksDeleteDetachEnum",
+        "modelAsString": true
+      }
+    },
+    "DeploymentStackValidateResult": {
+      "type": "object",
+      "description": "The Deployment stack validation result.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureResourceBase"
+        },
+        {
+          "$ref": "#/definitions/DeploymentStacksError"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/DeploymentStackValidateProperties",
+          "description": "The validation result details."
+        }
+      }
+    },
+    "DeploymentStackValidateProperties": {
+      "type": "object",
+      "description": "The Deployment stack validation result details.",
+      "properties": {
+        "actionOnUnmanage": {
+          "$ref": "#/definitions/ActionOnUnmanage",
+          "description": "Defines the behavior of resources that are no longer managed after the Deployment stack is updated or deleted."
+        },
+        "correlationId": {
+          "type": "string",
+          "description": "The correlation id of the Deployment stack validate operation. It is in GUID format and is used for tracing.",
+          "example": "01234567-89ab-cdef-0000-000000000000"
+        },
+        "denySettings": {
+          "$ref": "#/definitions/DenySettings",
+          "description": "The Deployment stack deny settings."
+        },
+        "deploymentScope": {
+          "type": "string",
+          "description": "The Deployment stack deployment scope."
+        },
+        "description": {
+          "type": "string",
+          "description": "The Deployment stack validation description."
+        },
+        "parameters": {
+          "type": "object",
+          "description": "Deployment parameters."
+        },
```

Comment:
Free form objects are not allowed in ARM .  Can you define this correctly using types. ?


If this is the same as anARM template parameter then you can use the schema of defining this as a hashtoken and have only the "value" be freeform.

Reply:
Note for future reviewers all other LintDiff errors seem to be false negatives : https://github.com/Azure/azure-rest-api-specs/pull/27529/checks?check_run_id=24038093109

Reply:
https://github.com/Azure/azure-rest-api-specs/pull/27529/commits/1644d1e622fe02cadf13c98cda3a93cdb464d6f7